pinot-controller/src/main/java/org/apache/pinot/controller/workload/QueryWorkloadManager.java

```diff
@@ -58,108 +84,237 @@ public QueryWorkloadManager(PinotHelixResourceManager pinotHelixResourceManager)
+        nodeConfig.getPropagationScheme().getPropagationType());
+    for (PropagationEntity entity : nodeConfig.getPropagationScheme().getPropagationEntities()) {
+      if (entity.getOverrides() != null && propagationScheme.isOverrideSupported(entity)) {
+        List<PropagationEntityOverrides> overrides = entity.getOverrides();
+        // Apply each override separately and aggregate the instance costs
+        for (PropagationEntityOverrides override : overrides) {
+          resolveAndAggregateInstanceCosts(propagationScheme, entity, override, nodeConfig.getNodeType(),
+              instanceCostMap);
+        }
+      } else {
+        resolveAndAggregateInstanceCosts(propagationScheme, entity, null, nodeConfig.getNodeType(),
+            instanceCostMap);
+      }
     }
   }
 
+  private void resolveAndAggregateInstanceCosts(PropagationScheme propagationScheme,
+                                                PropagationEntity entity, PropagationEntityOverrides override,
+                                                NodeConfig.Type nodeType,
+                                                Map<String, InstanceCost> workloadInstanceCostMap) {
+    Set<String> instances = propagationScheme.resolveInstances(entity, nodeType, override);
+    Map<String, InstanceCost> entityInstanceCostMap = _costSplitter.computeInstanceCostMap(entity.getCpuCostNs(),
+        entity.getMemoryCostBytes(), instances);
+    PropagationUtils.mergeCosts(workloadInstanceCostMap, entityInstanceCostMap);
+  }
+
   /**
-   * Propagate delete workload refresh message for the given queryWorkloadConfig
-   * @param queryWorkloadConfig The query workload configuration to delete
-   * 1. Resolve the instances based on the node type and propagation scheme
-   * 2. Send the {@link QueryWorkloadRefreshMessage} with DELETE_QUERY_WORKLOAD_MSG_SUB_TYPE to the instances
+   * Propagates a delete for the given workload to all relevant instances.
+   *
+   * <p>
+   * The method resolves the target instances for each {@link NodeConfig} and sends a
+   * {@link QueryWorkloadRefreshMessage} with subtype
+   * {@link QueryWorkloadRefreshMessage#DELETE_QUERY_WORKLOAD_MSG_SUB_TYPE},
+   * which instructs the instance to remove local state associated with the workload and stop enforcing costs for it.
+   * </p>
+   *
+   * @param queryWorkloadConfig The workload to delete (only the name and node scoping are used).
    */
   public void propagateDeleteWorkloadMessage(QueryWorkloadConfig queryWorkloadConfig) {
     String queryWorkloadName = queryWorkloadConfig.getQueryWorkloadName();
-    for (NodeConfig nodeConfig: queryWorkloadConfig.getNodeConfigs()) {
-      Set<String> instances = resolveInstances(nodeConfig);
-      if (instances.isEmpty()) {
-        String errorMsg = String.format("No instances found for Workload: %s", queryWorkloadName);
-        LOGGER.warn(errorMsg);
-        continue;
+    LOGGER.info("Propagating workload delete for: {}", queryWorkloadName);
```

Comment:
Let's say I initially have a workload defined with Table PropagationType where Table1 and Table2 are the entities (assume no hosts are common between table1 and table2)

Now if I update the workload to  remove Table2 from the propagation entities, do we handle it by sending a deleteWorkloadMessage to hosts of table1?